### PR TITLE
HIVE-28315: Missing classes while using hive jdbc standalone jar

### PR DESCRIPTION
**Pull request by dengzhhu653**
### What changes were proposed in this pull request?
Add classes/trim dependencies to/from the jdbc standalone jar

### Why are the changes needed?
- Solve the missing classes
- Reduce the jar size from 98M to 42M

### Does this PR introduce _any_ user-facing change?
No

### Is the change a dependency upgrade?
No

### How was this patch tested?
  - Tested create/show/select/load queries, fetching the result;
  - Tested DatabaseMetaData;
  - Tested Kerberos & Delegation token & Simple authentication;
  - Tested using ZK for service discovery;
  - Inspected the classes inside the standalone jar.

### CODE REVIEW

File: jdbc/pom.xml [hunk 1]
```diff
@@ -189,17 +235,24 @@
...
+                        <include>org/apache/hadoop/fs/impl/*</include>
+                        <include>org/apache/hadoop/fs/statistics/*</include>
+                        <include>org/apache/hadoop/metrics2/**</include>
+                        <include>org/apache/hadoop/security/token/**</include>
+                        <include>org/apache/hadoop/util/*</include>
                         <include>org/apache/hadoop/io/Writable.class</include>
-                        <include>org/apache/hadoop/io/DataOutputBuffer.class</include>
-                        <include>org/apache/hadoop/io/DataInputBuffer.class</include>
+                        <include>org/apache/hadoop/net/*</include>
+                        <include>org/apache/hadoop/io/*</include>
```
Comment (deniskuzZ) [inline jdbc/pom.xml:245]: should we remove `org/apache/hadoop/io/Writable.class` above?

File: jdbc/pom.xml [hunk 2]
```diff
@@ -308,21 +369,27 @@
...
                       <exclude>org.codehaus.jettison:*</exclude>
                       <exclude>org.datanucleus:*</exclude>
+                      <exclude>org.eclipse.jetty:*</exclude>
                       <exclude>org.eclipse.jetty.aggregate:*</exclude>
+                      <exclude>org.ehcache:*</exclude>
                       <exclude>org.fusesource.leveldbjni:*</exclude>
                       <exclude>org.glassfish.jersey*:*</exclude>
                       <exclude>org.hamcrest:*</exclude>
                       <exclude>org.htrace:*</exclude>
+                      <exclude>org.jetbrains.kotlin:*</exclude>
```
Comment (deniskuzZ) [inline jdbc/pom.xml:392]: what is this `jetbrains.kotlin`?

File: jdbc/pom.xml [hunk 3]
```diff
@@ -189,17 +235,24 @@
...
                         <include>org/apache/hadoop/io/Writable.class</include>
-                        <include>org/apache/hadoop/io/DataOutputBuffer.class</include>
-                        <include>org/apache/hadoop/io/DataInputBuffer.class</include>
+                        <include>org/apache/hadoop/net/*</include>
+                        <include>org/apache/hadoop/io/*</include>
                         <include>org/apache/hadoop/HadoopIllegalArgumentException.class</include>
                       </includes>
                     </filter>
+                    <filter>
+                      <artifact>org.apache.hadoop:hadoop-auth</artifact>
```
Comment (deniskuzZ) [inline jdbc/pom.xml:250]: previously it was excluded completely, why it that change?

Comment (dengzhhu653) [reply jdbc/pom.xml:250]: This is for UGI to login the user from keytab, 
```
Exception in thread "main" java.lang.NoClassDefFoundError: org/apache/hadoop/util/PlatformName
	at org.apache.hadoop.security.UserGroupInformation.getOSLoginModuleName(UserGroupInformation.java:434)
	at org.apache.hadoop.security.UserGroupInformation.<clinit>(UserGroupInformation.java:462)
	at org.example.App.main(App.java:45)

File: jdbc/pom.xml (hunk 2 above)
Comment (dengzhhu653) [reply jdbc/pom.xml:392]: It's introduced by hadoop-hdfs-client, 
```
[INFO] |  |  +- org.apache.hadoop:hadoop-hdfs-client:jar:3.3.6:compile
[INFO] |  |  |  +- com.squareup.okhttp3:okhttp:jar:4.9.3:compile
[INFO] |  |  |  |  \- com.squareup.okio:okio:jar:2.8.0:compile
[INFO] |  |  |  +- org.jetbrains.kotlin:kotlin-stdlib:jar:1.4.10:compile
[INFO] |  |  |  \- org.jetbrains.kotlin:kotlin-stdlib-common:jar:1.4.10:compile

**Review by deniskuzZ — APPROVED** [review]
LGTM +1

**Pull request merged**